apps/base-docs/docs/pages/identity/smart-wallet/guides/linked-accounts/setup.mdx

```diff
@@ -0,0 +1,237 @@
+
+```
+app/
+   layout.tsx
+   page.tsx
+```
+
+## Setting up the Coinbase Wallet Context
+
+Create a new directory called `context` under the `app` directory and add a React Context for the Coinbase SDK. This will make it easier to interact with the wallet throughout your application.
+
+```tsx
+// app/context/CoinbaseWalletContext.tsx
+'use client';
+import { createCoinbaseWalletSDK, ProviderInterface } from '@coinbase/wallet-sdk';
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+  ReactNode,
+  useMemo,
+  useCallback,
+} from 'react';
+import { Address, createWalletClient, custom } from 'viem';
+import { baseSepolia } from 'viem/chains';
+
+interface CoinbaseWalletContextType {
+  provider: ProviderInterface | null;
+  connect: () => Promise<void>;
+  disconnect: () => void;
+  isConnected: boolean;
+  address: Address | null;
+}
+
+const CoinbaseWalletContext = createContext<CoinbaseWalletContextType>({
+  provider: null,
+  connect: async () => {},
+  disconnect: () => {},
+  isConnected: false,
+  address: null,
+});
+
+export function CoinbaseWalletProvider({ children }: { children: ReactNode }) {
+  const [provider, setProvider] = useState<ProviderInterface | null>(null);
+  const [isConnected, setIsConnected] = useState(false);
+  const [address, setAddress] = useState<Address | null>(null);
+  useEffect(() => {
+    // Initialize Coinbase Wallet SDK
+    const coinbaseWalletSDK = createCoinbaseWalletSDK({
+      appName: 'Coinbase Wallet demo',
+      appChainIds: [baseSepolia.id],
+      preference: {
+        options: 'smartWalletOnly',
+        keysUrl: 'https://keys-dev.coinbase.com/connect',
+      },
+    });
+    setProvider(coinbaseWalletSDK.getProvider());
+  }, []);
+  const walletClient = useMemo(() => {
+    if (!provider) return null;
+    return createWalletClient({
+      chain: baseSepolia,
+      transport: custom({
+        async request({ method, params }) {
+          const response = await provider.request({ method, params });
+          return response;
+        },
+      }),
+    });
+  }, [provider]);
+
+  const connect = useCallback(async () => {
+    if (!walletClient || !provider) return;
+    walletClient.requestAddresses().then(async (addresses) => {
+      if (addresses.length > 0) {
+        setAddress(addresses[0]);
+        setIsConnected(true);
+      }
+    });
+  }, [walletClient, provider]);
+
+  const disconnect = () => {
+    if (!provider) return;
+    try {
+      provider.disconnect();
+      setIsConnected(false);
+      setAddress(null);
+    } catch (error) {
+      console.error('Failed to disconnect from Coinbase Wallet:', error);
+    }
+  };
+
+  return (
+    <CoinbaseWalletContext.Provider
+      value={{
+        provider,
+        connect,
+        disconnect,
+        isConnected,
+        address,
+      }}
+    >
+      {children}
+    </CoinbaseWalletContext.Provider>
+  );
+}
+
+export function useCoinbaseWallet() {
```

Comment:
do we think most developers will want to make this custom hook? Or will just want to use us via a wallet aggregator with minimal diffs @montycheese @cb-jake

Reply:
>  Or will just want to use us via a wallet aggregator with minimal diffs
this is most likely (or using us via OnchainKit) but we need to figure out what our approach is to getting the developer access to our SDK first. Or look into viem extensions

Reply:
We aligned on providing a way for people to use the new RPC via wagmi hooks + viem functions